From a266d42863e2bbca045275c684c5ad76281bbf5f Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 18:35:12 -0400
Subject: [ethernet] bnx2: Don't build unused suspend/resume functions not enabled

Message-id: <1409942115-26825-6-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91498
O-Subject: [RHEL7.1 PATCH 5/8] bnx2: Don't build unused suspend/resume functions not enabled
Bugzilla: 1092495
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1092495

Upstream commit:
commit 77d149c4eb8964b6bd4a929b102a867505add612
Author: Daniel J Blueman <daniel@quora.org>
Date:   Fri Apr 11 16:14:26 2014 +0800

    bnx2: Don't build unused suspend/resume functions not enabled

    When CONFIG_PM_SLEEP isn't enabled, bnx2_suspend/resume are unused; don't
    build them when they aren't used.

    Signed-off-by: Daniel J Blueman <daniel@quora.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2.c b/drivers/net/ethernet/broadcom/bnx2.c
index a8efb18..0ab8370 100644
--- a/drivers/net/ethernet/broadcom/bnx2.c
+++ b/drivers/net/ethernet/broadcom/bnx2.c
@@ -8627,6 +8627,7 @@ bnx2_remove_one(struct pci_dev *pdev)
  pci_disable_device(pdev);
 }
 
+#ifdef CONFIG_PM_SLEEP
 static int
 bnx2_suspend(struct device *device)
 {
@@ -8665,7 +8666,6 @@ bnx2_resume(struct device *device)
  return 0;
 }
 
-#ifdef CONFIG_PM_SLEEP
 static SIMPLE_DEV_PM_OPS(bnx2_pm_ops, bnx2_suspend, bnx2_resume);
 #define BNX2_PM_OPS (&bnx2_pm_ops)
 
-- 
1.7.1